src/modules/core/components/Fields/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,252 @@
+import { useField } from 'formik';
+import React, { useCallback } from 'react';
+import { defineMessages, MessageDescriptor, useIntl } from 'react-intl';
+import ReactDatePicker, { CalendarContainer } from 'react-datepicker';
+import classnames from 'classnames';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+import Icon from '~core/Icon';
+
+import TimePicker from './TimePicker';
+import {
+  DEFAULT_DATE_FORMAT,
+  DEFAULT_TIME_FORMAT,
+  MONTH_NAMES,
+} from './constants';
+import styles from './DatePicker.css';
+
+const MSG = defineMessages({
+  expandIconHTMLTitle: {
+    id: 'DatePicker.expandIconHTMLTitle',
+    defaultMessage: 'expand',
+  },
+});
+
+interface DatePickerFieldValue {
+  date: Date;
+  option?: any;
+}
+
+export interface DatePickerOption {
+  label: string | MessageDescriptor;
+  value: any;
```

Comment:
How come value is `any` here instead of `string`?

Reply:
That's an interesting one as I didn't want to restrict the type of the option's value. Many HTML elements (button, input) accept a `value` attribute which is of type `string | number | string[] | undefined` so I suppose I'll change my typing accordingly